test/sorting.jl

```diff
@@ -83,6 +85,25 @@ end
 @test searchsorted([1,2,3], 0) == 1:0
 @test searchsorted([1,2,3], 4) == 4:3
 
+immutable ConstantRange{T} <: Range{T}
```

Comment:
This section would benefit from a comment. (I know lots of others would too,  but it is easier to complain in a not-yet-merged PR) 

Eg:
```julia
# exercise the codepath in searchsorted* methods for ranges that check for zero step range
```